src/main/scala/rps/Game.scala

```diff
@@ -1,57 +1,37 @@
 package rps
 
-object Move extends Enumeration {
-  type Move = Value
-  val Rock = Value("0")
-  val Paper = Value("1")
-  val Scissor = Value("2")
-
-  def isOrderType(s: String) = values.exists(_.toString == s)
-}
-
-object Result extends Enumeration {
-  type Result = Value
-  val Won = Value("you won!")
-  val Lost = Value("you lost...")
-  val Tie = Value("it's a tie!")
-}
+import scala.util.Random
+import model.{Move, Result}
+import Move._, Result._
 
 object Game {
-  def getRandomMove(): Move.Move = {
-    Move(scala.util.Random.nextInt(Move.maxId))
-  }
-
-  def getMoveName(move: Move.Move): String = move match {
-    case Move.Rock    => "Rock"
-    case Move.Paper   => "Paper"
-    case Move.Scissor => "Scissor"
-  }
-
-  def getGameResult(a: Move.Move, b: Move.Move): Result.Result =
+  def getGameResult(a: Move)(b: Move): Result =
     (a, b) match {
-      case (a, b) if a == b => Result.Tie
-      case (Move.Rock, Move.Paper) | (Move.Paper, Move.Scissor) |
-          (Move.Scissor, Move.Rock) =>
-        Result.Won
-      case _ => Result.Lost
+      case (a, b) if a == b => Tie
+      case (Rock, Paper) | (Paper, Scissors) | (Scissors, Rock) =>
+        Won
+      case _ => Lost
     }
 
   def play(): Unit = {
     println("make your move (0 for Rock, 1 for Paper, 2 for Scissors):")
 
-    val userMove = scala.io.StdIn.readLine()
+    val userInput = scala.io.StdIn.readLine()
+
+    val maybeUserMove = Move.moves.find(move => Move.getId(move) == userInput)
+
+    maybeUserMove match {
+      case Some(userMove: Move) => {
+        val ourMove = Random.shuffle(Move.moves).head
+
+        println(s"you played ${Move.getName(userMove)}")
+        println(s"we played ${Move.getName(ourMove)}")
 
-    Move.isOrderType(userMove) match {
-      case true => {
-        val userMoveName = getMoveName(Move.withName(userMove))
-        val ourMove = getRandomMove()
-        val ourMoveName = getMoveName(ourMove)
+        val getFinalResult = Result.getMessage _ compose getGameResult(ourMove)
```

Comment:
I was sure to have commented on this before, but I can't find the comment anymore.

Anyway, I like the spirit of what you're doing here and it's good that you're experimenting with function composition, but the final result is borderline unreadable :)

A similar but more readable alternative would be

```scala
val getFinalResult = (getGameResult(ourMove) _) andThen Result.getMessage
```

but I will never find it more readable than

```scala
val result = getGameResult(ourMove, userMove)
println(Result.getMessage(ourMove))
```

or even

```scala
println(Result.getMessage(getGameResult(ourMove, userMove)))
```

Reply:
I perfectly agree, here function composition is useless (as well as painfully difficult to read due to the "method to function" conversion..). As you pointed out it was an experiment to gain confidence with scala sintax and constructs. I'll revert to a more readable implementation 👍 
(p.s. it also took me a long time to end up with a working implementation, @dhinus helped me with it but there are a couple unanswered technical questions I would like to ask you if you have a spare 5 minutes...)

Reply:
I have 5 minutes, but on appear.in/gabro if you want (I'm out of office today)